### PR TITLE
Add CleanupRemoteStorages command

### PR DESCRIPTION
**Pull request by MorrisJobke**
Remote storages with "shared::" never get deleted. This becomes a problem if a user has reshared an incoming remote share that has been deleted, because that internal re share never gets deleted (fileid still exists in filecache and storage for that file is never deleted). The reshare recipient now gets an errer when trying to access that file/folder.

* downstream of https://github.com/owncloud/core/pull/26379

### CODE REVIEW

File: apps/files_sharing/lib/Command/CleanupRemoteStorages.php [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+		$output->writeln(count($remoteStorages) . " remote storage(s) need(s) to be checked");
+
+		$remoteShareIds = $this->getRemoteShareIds();
+
+		$output->writeln(count($remoteShareIds) . " remote share(s) exist");
+
+		foreach ($remoteShareIds as $id => $remoteShareId) {
+			if (isset($remoteStorages[$remoteShareId])) {
+				$output->writeln("$remoteShareId belongs to remote share $id");
```
Comment (nickvergessen) [inline apps/files_sharing/lib/Command/CleanupRemoteStorages.php]: only print this on `--verbose`?
I'd also use colored output, to make clear whether this or the line below is good/bad

**Review by rullzer — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]
Added two styling commits

**Comment by nickvergessen** [comment]
I wonder if that should be run as a repairstep as well? Or might it be too long running?

**Comment by MorrisJobke** [comment]
@nickvergessen: maybe a weekly background job would be nice

**Comment by nickvergessen** [comment]
Create a starter issue?

**Comment by MorrisJobke** [comment]
> Create a starter issue?

#3903

**Pull request merged**